src/v/compat/tests/compat_test.cc

```diff
@@ -60,7 +62,9 @@ perturb(json::Value& value, json::Document& doc, std::string& path) {
             }
         }
         const auto orig = v;
-        const int modifier = v == 0 ? 1 : -1;
+        const int modifier = (v == 0 || v == std::numeric_limits<T>::min())
+                               ? 1
+                               : -1;
```

Comment:
can you explain this one? i don't see anything in the original expression that would be problematic. presumably T was unsigned? was the problem on the next statement?

Reply:
Yep, `v (std::numeric_limits<T>::min()) += modifier (-1);` underflows and triggers UBSAN diagnostic